src/app/components/auth/signup/signup.component.spec.ts

```diff
@@ -33,10 +34,63 @@ describe('SignupComponent', () => {
   beforeEach(() => {
     fixture = TestBed.createComponent(SignupComponent);
     component = fixture.componentInstance;
-    fixture.detectChanges();
   });
 
   it('should create', () => {
+    fixture.detectChanges();
     expect(component).toBeTruthy();
   });
+
+
+  it('should call userSignup successfully',  inject([ApiService], ( service: ApiService) => {
+      spyOn(service, 'postUrl').and.returnValue(new Observable((observation) => {
+        observation.next({'status': 201});
+        observation.complete();
+        return{unsubscribe() {}};
```

Comment:
I am seeing spacing being off in some places in this PR.
```suggestion
        return {unsubscribe() {}};
```